src/stk500v2.c

```diff
@@ -3712,77 +3710,184 @@ static void stk500v2_display(const PROGRAMMER *pgm, const char *p) {
 }
 
 
-static void stk500v2_print_parms1(const PROGRAMMER *pgm, const char *p, FILE *fp) {
-  unsigned char vtarget = 0, vadjust = 0, osc_pscale = 0, osc_cmatch = 0, sck_duration =0; //XXX 0 is not correct, check caller
-  unsigned int sck_stk600, clock_conf, dac, oct, varef;
-  unsigned char vtarget_jtag[4];
+static double stk500v2_vtarget_value(const PROGRAMMER *pgm) {
+  if (PDATA(pgm)->pgmtype == PGMTYPE_JTAGICE_MKII) {
+    unsigned char vtarget_jtag[4];
+    memset(vtarget_jtag, 0, sizeof vtarget_jtag);
+    PROGRAMMER *pgmcp = pgm_dup(pgm);
+    pgmcp->cookie = PDATA(pgm)->chained_pdata;
+    jtagmkII_getparm(pgmcp, PAR_OCD_VTARGET, vtarget_jtag);
+    pgm_free(pgmcp);
+    return b2_to_u16(vtarget_jtag) / 1000.0;
+  } else if (PDATA(pgm)->pgmtype != PGMTYPE_JTAGICE3) {
```

Comment:
Elsing a return is a bit unusual. It's defo not needed. And code might look clear if a clear line is left
```
  if(...) {
    ...
    return ...;
  }

  if(...) {
    ...
    return ...;
  }

  return 0;
}
```

Reply:
fully agree :)